app/models/miq_server/configuration_management.rb

```diff
@@ -2,17 +2,12 @@ module MiqServer::ConfigurationManagement
   extend ActiveSupport::Concern
   include ConfigurationManagementMixin
 
-  def get_config(type = "vmdb")
-    if is_local?
-      VMDB::Config.new(type)
-    else
-      VMDB::Config.for_resource(type, self)
-    end
+  def settings
+    (is_local? ? ::Settings : Vmdb::Settings.for_resource(self)).to_hash
   end
 
-  def set_config(config)
-    config = config.config if config.respond_to?(:config)
-    add_settings_for_resource(config)
+  def settings=(new_settings)
```

Comment:
After looking at the UI PR, I'm wondering if we can just remove this method entirely and use `#add_settings_for_resource` as this method doesn't add any additional logic now.